src/compiler/core.ts

```diff
@@ -1787,14 +1787,21 @@ namespace ts {
      *        (0.4 allows 1 substitution/transposition for every 5 characters,
      *         and 1 insertion/deletion at 3 characters)
      */
-    export function getSpellingSuggestion<T>(name: string, candidates: T[], getName: (candidate: T) => string | undefined): T | undefined {

```

Comment:
Instead of disturbing this function and having to check for this all the time, i would recommend adding 
``getSpellingSuggestionForJsx` that before calling to this function checks if name is either for and candidate includes htmlFor or name is class and candidate has className instead.

Reply:
Hmm I have no clue about how to do this change because the call chain from the `checker` => `getSuggestionForNonexistentProperty` => `getSuggestedSymbolForNonexistentProperty` => `getSpellingSuggestionForName` => `getSpellingSuggestionForName` => `getSpellingSuggestion` spreads too far. If I make a separate function for JSX I have to re-implement all previous function logic for JSX to get the suggestion result.

Reply:
@sheetalkamat hello?